### ISSUE TITLE
Need Help - Logout OKTA

### ISSUE THREAD

**Issue by bibhukdas** [KentorIT/authservices#481]
We are using kentor to leverage OKTA (SAML)for authenticating users.
While logging out, we are calling a controller method, which clears the cookie and redirects to OKTA URL.

 var ctx = Request.GetOwinContext();
            var authManager = ctx.Authentication;
            authManager.SignOut("Cookies");

return Redirect("https://xxxxxx.oktapreview.com/login/signout");

Now, after this, the page authomatically redirects to okta login, which is fine. But once i login again, it goes to the OKTA dashboard instead of the app.

Can anyone give some idea to implement logout ?

Basically the flow should be click logout - goes to okta login page- log in to go to app.

**Comment by AndersAbel** [comment]
> But once i login again, it goes to the OKTA dashboard instead of the app.

Where do you initiate that second login? In your app, or on the Okta side?

**Comment by bibhukdas** [comment]
On the OKTA side.

**Comment by AndersAbel** [comment]
Then you are effectively doing an IDP-initiated Signin. At that point AuthServices as an SP is not involved at all. It is up to the OKTA Idp to decide where to redirect after successful sign in. So this is an OKTA issue and nothing for AuthServices.

**Comment by explunit** [comment]
@bibhukdas are you really just redirecting to https://xxxxxx.oktapreview.com/login/signout? I think you need to redirect with a signed SAML logout request. If you are using the OWIN middleware it looks like that is generated for you automatically:
https://github.com/KentorIT/authservices/blob/master/SampleOwinApplication/Controllers/AccountController.cs#L397
